### ISSUE TITLE
Build failed on Macbook M1

### ISSUE THREAD

**Issue by anarnoli** [NativeScript/NativeScript#9383]
**Environment**
- Machine: MacBook M1
 - CLI: 8
 - iOS Runtime: 14.5
 - XCode Version: 12.5

**Describe the bug**
I'm trying to setup Macbook M1 for Nativescript 8. I followed Environment Setup steps for MacOS + iOS. I also tried to follow "Additional notes for M1 based machines" recommendation:

softwareupdate --install-rosetta
_Successfully installed_

sudo arch -x86_64 gem install ffi 
Throws error _arch: posix_spawnp: gem: Bad CPU type in executable_

arch -x86_64 pod install
Throws error _arch: posix_spawnp: pod: Bad CPU type in executable_

I'm seeking help for setting up environment. When I'm trying to build HelloWorld Angular project it throws below error:

ashishnarnoli@Ashishs-MacBook-Air test % ns run ios 

Searching for devices...
Preparing project...
webpack is watching the files...
(node:11625) [DEP0148] DeprecationWarning: Use of deprecated folder mapping "./" in the "exports" field module resolution of the package at /Users/ashishnarnoli/test/node_modules/postcss/package.json.
Update this package.json to use a subpath pattern like "./*".
(Use `node --trace-deprecation ...` to show where the warning was created)
asset vendor.js 16.5 MiB [compared for emit] (name: vendor) (id hint: defaultVendor)
asset bundle.js 460 KiB [compared for emit] (name: bundle)
asset tns_modules/inspector_modules.js 90 KiB [compared for emit] (name: tns_modules/inspector_modules)
Entrypoint bundle 17 MiB = vendor.js 16.5 MiB bundle.js 460 KiB
Entrypoint tns_modules/inspector_modules 16.6 MiB = vendor.js 16.5 MiB tns_modules/inspector_modules.js 90 KiB
orphan modules 228 KiB [orphan] 132 modules
runtime modules 50.2 KiB 28 modules
modules by path ./node_modules/ 6.26 MiB
  javascript modules 6 MiB 374 modules
  json modules 273 KiB
    modules by path ./node_modules/mdn-data/css/*.json 242 KiB 3 modules
    modules by path ./node_modules/css-tree/ 30.7 KiB 2 modules
modules by path ./src/ 131 KiB
  modules by path ./src/app/ 9.35 KiB
    modules by path ./src/app/*.ts 3.54 KiB 3 modules
    modules by path ./src/app/item/*.ts 5.81 KiB 3 modules
  ./src/main.ts 3.54 KiB [built] [code generated]
  ./src/app.css 118 KiB [built] [code generated]
external "~/package.json" 42 bytes [optional] [built] [code generated]
./$$_lazy_route_resource/ lazy namespace object 160 bytes [built] [code generated]
webpack 5.31.2 compiled successfully in 10151 ms
Webpack compilation complete. Watching for file changes.
**Installing pods...
arch: posix_spawnp: pod: Bad CPU type in executable
'arch install' command failed.**

**ns doctor**
ashishnarnoli@Ashishs-MacBook-Air ~ % ns doctor
✔ Getting environment information 

No issues were detected.
✔ Your ANDROID_HOME environment variable is set and points to correct directory.
✔ Your adb from the Android SDK is correctly installed.
✔ The Android SDK is installed.
✔ A compatible Android SDK for compilation is found.
✔ Javac is installed and is configured properly.
✔ The Java Development Kit (JDK) is installed and is configured properly.
✔ Xcode is installed and is configured properly.
✔ xcodeproj is installed and is configured properly.
✔ CocoaPods are installed.
✔ CocoaPods update is not required.
✔ CocoaPods are configured properly.
✔ Your current CocoaPods version is newer than 1.0.0.
✔ Python installed and configured correctly.
✔ The Python 'six' package is found.
✔ Xcode version 12.5.0 satisfies minimum required version 10.
✔ Getting NativeScript components versions information...
✔ Component nativescript has 8.0.1 version and is up to date.

**Package.json**
{
  "name": "@nativescript/template-hello-world-ng",
  "main": "src/main.ts",
  "version": "8.0.5",
  "author": "NativeScript Team <oss@nativescript.org>",
  "description": "NativeScript Application",
  "license": "SEE LICENSE IN <your-license-filename>",
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "App_Resources",
    "hooks",
    "src",
    "tools",
    "!tools/assets",
    ".editorconfig",
    "references.d.ts",
    "tsconfig.json"
  ],
  "keywords": [
    "nativescript",
    "mobile",
    "angular",
    "{N}",
    "template"
  ],
  "repository": "<fill-your-repository-here>",
  "bugs": {
    "url": "https://github.com/NativeScript/NativeScript/issues"
  },
  "dependencies": {
    "@angular/animations": "~11.2.7",
    "@angular/common": "~11.2.7",
    "@angular/compiler": "~11.2.7",
    "@angular/core": "~11.2.7",
    "@angular/forms": "~11.2.7",
    "@angular/platform-browser": "~11.2.7",
    "@angular/platform-browser-dynamic": "~11.2.7",
    "@angular/router": "~11.2.7",
    "@nativescript/angular": "~11.8.0",
    "@nativescript/core": "~8.0.0",
    "@nativescript/theme": "~3.0.1",
    "reflect-metadata": "~0.1.13",
    "rxjs": "~6.6.7",
    "zone.js": "~0.11.4"
  },
  "devDependencies": {
    "@angular/compiler-cli": "~11.2.7",
    "@nativescript/android": "8.0.0",
    "@nativescript/ios": "8.0.0",
    "@nativescript/types": "~8.0.0",
    "@nativescript/webpack": "beta",
    "@ngtools/webpack": "~11.2.6",
    "typescript": "~4.0.0"
  },
  "private": "true",
  "readme": "NativeScript Application"
}

**Expected behavior**
HelloWorld sample app should run on iOS Emulator

**Comment by janoshrubos** [comment]
I've found a solution on Stack Overflow, i hope it will help: https://stackoverflow.com/a/66771694

These are the steps I did to get back to a working state (and to apply the M1 workarounds for ffi):

Uninstall Ruby with: brew uninstall ruby --force
Uninstall CocoaPods. At first, try to list all CocoaPods versions / components with gem list --local | grep cocoapods. Then uninstall them one by one, in my case:
`sudo gem uninstall cocoapods`
`sudo gem uninstall cocoapods-core`
`sudo gem uninstall cocoapods-downloader`

etc.

As a next step I was able to reinstall ffi and then CocoaPods with:
`sudo arch -x86_64 gem install ffi`
`sudo arch -x86_64 gem install cocoapods`

Once done, I could run pod install as expected. Note that I had the Terminal open all the time in standard mode - no Rosetta required.
